core-models/src/main/java/com/anytypeio/anytype/core_models/ObjectWrapper.kt

```diff
@@ -226,7 +249,11 @@ sealed class ObjectWrapper {
 
         val type: List<Id> get() = getValues(Relations.TYPE)
 
-        val isValid get() = map.containsKey(Relations.RELATION_KEY) && map.containsKey(Relations.ID)
+        val isValid get() =
+            map.containsKey(Relations.RELATION_KEY) && map.containsKey(Relations.ID)
+
+        val isValidToUse get() = isValid && isDeleted != true && isArchived != true && isHidden != true
```

Comment:
let's use notDeletedNorArchived or extract it into external function which would be used by external code